onnxruntime/core/session/environment.cc

```diff
@@ -42,6 +42,146 @@
 #include "orttraining/core/optimizer/graph_transformer_registry.h"
 #endif
 
+#include "onnx/defs/data_type_utils.h"
+#include "onnx/defs/function.h"
+#include "onnx/defs/math/utils.h"
+#include "onnx/defs/schema.h"
+#include "onnx/defs/tensor_proto_util.h"
+
+namespace ONNX_NAMESPACE {
+
+    template <typename T>
+    static T get_scalar_value_from_tensor(const ONNX_NAMESPACE::TensorProto* t) {
+    if (t == nullptr) {
+        return T{};
+    }
+
+    auto data_type = t->data_type();
+    switch (data_type) {
+        case ONNX_NAMESPACE::TensorProto::FLOAT:
+        return static_cast<T>(ONNX_NAMESPACE::ParseData<float>(t).at(0));
+        case ONNX_NAMESPACE::TensorProto::DOUBLE:
+        return static_cast<T>(ONNX_NAMESPACE::ParseData<double>(t).at(0));
+        case ONNX_NAMESPACE::TensorProto::INT32:
+        return static_cast<T>(ONNX_NAMESPACE::ParseData<int32_t>(t).at(0));
+        case ONNX_NAMESPACE::TensorProto::INT64:
+        return static_cast<T>(ONNX_NAMESPACE::ParseData<int64_t>(t).at(0));
+        default:
+        fail_shape_inference("Unsupported input data type of ", data_type);
+    }
+    }
+
+    void OverrideSTFT()
```

Comment:
Comment documenting why we need this and what it does at a high level would be good. 

Info on whether this is a short-term workaround and when it could be removed would also be helpful so we can clean it up asap.